UInput.cpp

```diff
@@ -60,9 +60,20 @@ int fpc_uinput_create(fpc_uinput_t *uinput)
 
     rc = ioctl(fd, UI_DEV_SETUP, &usetup);
     if (rc < 0) {
-        ALOGE("Failed to setup uinput device! rc=%d: %s", rc, strerror(errno));
-        close(fd);
-        return rc;
+        ALOGW("Failed to setup uinput device! rc=%d: %s, falling back to write.", rc, strerror(errno));
+
+        struct uinput_user_dev usetup_legacy;
+        memset(&usetup_legacy, 0, sizeof(usetup_legacy));
+        usetup_legacy.id.bustype = BUS_VIRTUAL;
```

Comment:
A designated initializer is preferred here, in part to stay consistent with the initializer for `struct uinput_setup usetup`.

```suggestion
        struct uinput_user_dev usetup_legacy = {
            .id.bustype = BUS_VIRTUAL,
        };
```